src/wallet/wallet2.cpp

```diff
@@ -6248,10 +6250,19 @@ bool wallet2::check_version(uint32_t *version, bool *wallet_is_outdated, bool *d
     return false;
   }
 
-  // check wallet compatibility with daemon's hard fork version
   if (!m_allow_mismatched_daemon_version)
+  {
+    if (rpc_version < MAKE_CORE_RPC_VERSION(3, 17))
+    {
+      // Wallet cannot init FCMP++ tree while syncing if pointing to an older daemon
+      *daemon_is_outdated = true;
```

Comment:
```suggestion
      if (daemon_is_outdated)
        *daemon_is_outdated = true;
```

Null-pointer check or segfault ensues

Reply:
Nice catch